website_src/docs/r/service_dynamic_snippet_content_v1.html.markdown.tmpl

```diff
@@ -59,51 +61,95 @@ Multiple dynamic snippets:
-   name    = "AWS S3 hosting"
-   port    = 80
- }
-
- dynamicsnippet {
-   name     = "My Dynamic Snippet One"
-   type     = "recv"
-   priority = 110
- }
- 
- dynamicsnippet {
-      name     = "My Dynamic Snippet Two"
-      type     = "recv"
-      priority = 110
-    }
+  name = "snippet_test"
+
+  domain {
+    name    = "snippet.fastlytestdomain.com"
+    comment = "snippet test"
+  }
+
+  backend {
+    address = "tftesting.tftesting.net.s3-website-us-west-2.amazonaws.com"
+    name    = "AWS S3 hosting"
+    port    = 80
+  }
+
+  dynamicsnippet {
+    name     = "My Dynamic Snippet One"
+    type     = "recv"
+    priority = 110
+  }
 
- default_host = "tftesting.tftesting.net.s3-website-us-west-2.amazonaws.com"
+  dynamicsnippet {
+       name     = "My Dynamic Snippet Two"
+       type     = "recv"
+       priority = 110
+     }
 
- force_destroy = true
+  default_host = "tftesting.tftesting.net.s3-website-us-west-2.amazonaws.com"
+
+  force_destroy = true
 }
 
 resource "fastly_service_dynamic_snippet_content_v1" "my_dyn_content_one" {
+  for_each = {
+    for d in fastly_service_v1.myservice.dynamicsnippet : d.name => d if d.name == "My Dynamic Snippet One"
+  }
 
- service_id = fastly_service_v1.myservice.id
- snippet_id = { for s in fastly_service_v1.myservice.dynamicsnippet : s.name => s.snippet_id }["My Dynamic Snippet One"]
+  service_id = fastly_service_v1.myservice.id
+  snippet_id = each.value.snippet_id
 
- content = "if ( req.url ) {\n set req.http.my-snippet-test-header-one = \"true\";\n}"
+  content = "if ( req.url ) {\n set req.http.my-snippet-test-header-one = \"true\";\n}"
 
 }
 
 resource "fastly_service_dynamic_snippet_content_v1" "my_dyn_content_two" {
+  for_each = {
+    for d in fastly_service_v1.myservice.dynamicsnippet : d.name => d if d.name == "My Dynamic Snippet Two"
+  }
 
- service_id = fastly_service_v1.myservice.id
- snippet_id = { for s in fastly_service_v1.myservice.dynamicsnippet : s.name => s.snippet_id }["My Dynamic Snippet Two"]
+  service_id = fastly_service_v1.myservice.id
+  snippet_id = each.value.snippet_id
 
- content = "if ( req.url ) {\n set req.http.my-snippet-test-header-two = \"true\";\n}"
+  content = "if ( req.url ) {\n set req.http.my-snippet-test-header-two = \"true\";\n}"
+
+}
+```
+
+## Example Usage (Terraform >= 0.12.0 && < 0.12.6)
+
+`for_each` attributes were not available in Terraform before 0.12.6, however, users can still use `for` expressions to achieve
+similar behaviour as seen in the example below.
+
+~> **Warning:** Terraform might not properly calculate implicit dependencies on computed attributes when using `for` expressions
+
+For scenarios such as adding a Dynamic Snippet to a service and at the same time, creating the Dynamic Snippets (`fastly_service_dynamic_snippet_content_v1`)
+resource, Terraform will not calculate compute implicit dependencies correctly on `for` expressions. This will result in index lookup
```

Comment:
I think this was a typo as "calculate compute" sounds weird, but it may have been intentional? I went with "calculate" to be consistent with the warning sentence above. 

```suggestion
resource, Terraform will not calculate implicit dependencies correctly on `for` expressions. This will result in index lookup
```